tox/exc.py

```diff
@@ -0,0 +1,77 @@
+import os
+import signal
+
+
+def exit_code_str(exception_name, command, exit_code):
+    """String representation for an InvocationError, with exit code
+
+    NOTE: this might also be used by plugin tests (tox-venv at the time of writing),
+    so some coordination is needed, if this is ever moved or a different solution for this hack
+    is found.
+
+    NOTE: this is aseparate function because pytest-mock `spy` does not work on Exceptions
+    We can use neither a class method nor a static because of https://bugs.python.org/issue23078.
+    Even a normal method failed with "TypeError: descriptor '__getattribute__' requires a
+    'BaseException' object but received a 'type'".
+    """
+    str_ = "%s for command %s" % (exception_name, command)
+    if exit_code is not None:
+        str_ += " (exited with code %d)" % (exit_code)
+        if (os.name == 'posix') and (exit_code > 128):
+            signals = {number: name
+                       for name, number in vars(signal).items()
+                       if name.startswith("SIG")}
+            number = exit_code - 128
+            name = signals.get(number)
+            if name:
+                str_ += ("\nNote: this might indicate a fatal error signal "
+                         "({} - 128 = {}: {})".format(number+128, number, name))
+    return str_
+
+
+class exception:
```

Comment:
if we use this as a namespace, wouldn't be nicer to make it a module, and use it as a file instead? this class thingy breaks PEP-8 and seems un-pythonic

Reply:
That is definitely much saner than what I did. Then we would have an `exception` module and the naming thing is also sorted.